src/Documentation/grains/transactions.md

```diff
@@ -88,31 +116,54 @@ public class AccountGrain : Grain, IAccountGrain
     private readonly ITransactionalState<Balance> balance;
 
     public AccountGrain(
-        [TransactionalState("balance")] ITransactionalState<Balance> balance)
+        [TransactionalState("balance", "TransactionStore")]
+        ITransactionalState<Balance> balance)
     {
         this.balance = balance ?? throw new ArgumentNullException(nameof(balance));
     }
 
     Task IAccountGrain.Deposit(uint ammount)
     {
-        this.balance.State.Value += ammount;
-        this.balance.Save();
-        return Task.CompletedTask;
+        return this.balance.PerformUpdate(x => x.Value += amount);
     }
 
     Task IAccountGrain.Withdrawal(uint ammount)
     {
-        this.balance.State.Value -= ammount;
-        this.balance.Save();
-        return Task.CompletedTask;
+        return this.balance.PerformUpdate(x => x.Value -= amount);
     }
 
     Task<uint> IAccountGrain.GetBalance()
     {
-        return Task.FromResult(this.balance.State.Value);
+        return this.balance.PerformRead(x => x.Value);
     }
-}
+}`
+```
+
+In the above example, the attribute `TransactionalState` is used to declare that the ‘balance’ constructor argument should be associated with a transactional state named “balance”.
+With this declaration, Orleans will wire up an `ITransactionalState` instance with a state loaded from the transactional state storage named "TransactionStore" (see Setup).
+The state can be modified via `PerformUpdate` or read via `PerformRead`.
+The transaction infrastructure will ensure that any such changes performed as part of a transaction, even among multiple grains distributed over an Orleans cluster, will either all be committed or reverted upon completion of the grain call which created the transaction (`IATMGrain.Transfer` in the above examples).
+
+### Calling Transactions
+
+Transactional methods on a grain interface are called the same as any other grain call.
+
+``` csharp
+    IATMGrain atm = client.GetGrain<IATMGrain>(0);
+    Guid from = Guid.NewGuid();
+    Guid to = Guid.NewGuid();
+    await atm.Transfer(from, to, 100);
+    uint fromBalance = await client.GetGrain<IAccountGrain>(from).GetBalance();
```

Comment:
nit: it's awkward to me to have a `uint` for balance, I would expect a `decimal` or a signed `int` (eg, if the currency is non-divisible tokens of some sort). Recommend `int`.

Overdrawing an account (having a negative balance) is normal.

This assumes we don't have validation logic (see above comment)

Reply:
I'm happy to change this if it's okay with @jason-bragg